### ISSUE TITLE
I encountered a problem when remotely connecting to OPCServer

### ISSUE THREAD

**Issue by lishuaia** [huskar-t/opcda#12]
I encountered a problem when remotely connecting to OPCServer;
MatrikonOPC can connect remotely, but cannot connect using your library,serverlist report errors：server list EnumClassesOfCategories: Access is denied.Could you please help me answer this question? Thank you

The following image shows MatrikonOPC connected to a remote OPCServer
![1724319808877](https://github.com/user-attachments/assets/0d861ccf-1110-40d5-ad07-3e1413a2e93e)

The following image shows the error message of the server list
![Uploading 1724319939918.jpg…]()

**Comment by lishuaia** [comment]
The image of the code was not uploaded successfully. This is the running result image of the code
![1724319939918](https://github.com/user-attachments/assets/5fbcf107-facf-4d28-a153-fe148328cb58)

**Comment by huskar-t** [comment]
@lishuaia Is there a problem connecting to localhost to execute GetOPCServer？

**Comment by huskar-t** [comment]
@lishuaia Can you test using this version?
`github.com/huskar-t/opcda v0.3.1-0.20240822113317-9cd33d1b0b09`

**Comment by lishuaia** [comment]
localhost no problem, The remote connection is down发自我的 iPhone在 2024年8月22日，19:51，Xuefeng Tan ***@***.***> 写道：﻿
@lishuaia Can you test using this version?
github.com/huskar-t/opcda v0.3.1-0.20240822113317-9cd33d1b0b09

**Comment by huskar-t** [comment]
> localhost no problem, The remote connection is down发自我的 iPhone在 2024年8月22日，19:51，Xuefeng Tan ***@***.***> 写道：﻿ @lishuaia Can you test using this version? github.com/huskar-t/opcda v0.3.1-0.20240822113317-9cd33d1b0b09 —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

@lishuaia Is there still a problem with using this version, and are there any error messages?

**Comment by lishuaia** [comment]
> @lishuaia Can you test using this version? `github.com/huskar-t/opcda v0.3.1-0.20240822113317-9cd33d1b0b09`

I couldn't find the code for this version? The tag on GitHub only has code for version v0.3.0

**Comment by huskar-t** [comment]
@lishuaia modify go.mod file 
![image](https://github.com/user-attachments/assets/583c97eb-5d7b-416f-9113-5551587ad5f4)

**Comment by lishuaia** [comment]
> v0.3.1-0.20240822113317-9cd33d1b0b09

我测试了下，还是报错：Access is denied

**Comment by huskar-t** [comment]
@lishuaia 
Could you confirm which line in the opcda library returns the error?

**Comment by lishuaia** [comment]
@huskar-t 
The specific error is in the enum ClassesOfCategories method of IOPCServerList.go
![1724642417050](https://github.com/user-attachments/assets/ebd25432-4a22-4eb0-85fd-530b127fca6f)

**Comment by huskar-t** [comment]
@lishuaia It seems that you didn't replace the version I gave.
check your go.mod file and check if there is a method called `getServersFromReg` in opcserver.go

**Comment by lishuaia** [comment]
@huskar-t 
This is the code I downloaded locally. There is a getServersFromReg method in opcserver. go
![1724644583676](https://github.com/user-attachments/assets/180ec9d0-74f7-4fc2-b27c-469c0dcba9f1)

**Comment by huskar-t** [comment]
@lishuaia That's strange. In the GetOPCServers method, if the call to EnumClassesOfCategories fails, it will try to get it from the registry.Are you sure the error returned is caused by EnumClassesOfCategories?

**Comment by lishuaia** [comment]
@huskar-t 
Yes, I added printing in the code. The following image shows the printed code and the running result
![1724658198796](https://github.com/user-attachments/assets/8d092382-1f22-4fe7-a184-5e798c0609f4)
![1724658171578](https://github.com/user-attachments/assets/314b8215-a888-411f-acf3-48e9011eb8e2)

**Comment by huskar-t** [comment]
> @huskar-t Yes, I added printing in the code. The following image shows the printed code and the running result ![1724658198796](https://private-user-images.githubusercontent.com/34432102/361352910-8d092382-1f22-4fe7-a184-5e798c0609f4.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2NTk2ODcsIm5iZiI6MTcyNDY1OTM4NywicGF0aCI6Ii8zNDQzMjEwMi8zNjEzNTI5MTAtOGQwOTIzODItMWYyMi00ZmU3LWExODQtNWU3OThjMDYwOWY0LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI2VDA4MDMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZmYjRhN2E5ZDBmZTliNzk4ZWM5YjUxMzM2ODVmYTgwNzZjMmNlYzQ4ZGUzZTQ1OTBmNTdlYjk5YmJkODQyYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3EMDbHpBrsZk-Oueq4VIswxpHtYplzZI3rqjmuj3cUY) ![1724658171578](https://private-user-images.githubusercontent.com/34432102/361352925-314b8215-a888-411f-acf3-48e9011eb8e2.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2NTk2ODcsIm5iZiI6MTcyNDY1OTM4NywicGF0aCI6Ii8zNDQzMjEwMi8zNjEzNTI5MjUtMzE0YjgyMTUtYTg4OC00MTFmLWFjZjMtNDhlOTAxMWViOGUyLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI2VDA4MDMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZDI0ZDk5ODBlOTkyOGJiMTU1MDFmN2ZjZDBkZWFjOTU2Y2IwNDdhMWJkZGVhNGM0NzZkNmViMjg5YTUxM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aUHICglyvQLuIJW8ivAq2RnMXpFVl4uOwai54tRvrk4)

From the screenshot, the error occurs in getServersFromReg

**Comment by huskar-t** [comment]
@lishuaia 加我 qq 吧，有时间我帮你远程看一下

**Comment by lishuaia** [comment]
@huskar-t 加您了，我的QQ名称是**迁就**，麻烦通过下

**Comment by giter** [comment]
+1 for the same problem.  I'm sure that the connection is successful when using opc-quick-client or konimarti/opc

**Comment by huskar-t** [comment]
@giter Can you try 0.3.3 and provide the call stack?

**Comment by giter** [comment]
@huskar-t `get clsid: getClsIDFromReg: Access is denied.`

`
    host := "LACLIENT2"
    progID := "ArchestrA.FSGateway.3"

    server, err := opcda.Connect(progID, host)
    if err != nil {
        panic(err)
    }
`

**Comment by huskar-t** [comment]
@giter Looks like there is a problem with the dcom configuration, please check the permissions and dcom configuration.
https://www.bilibili.com/video/BV1ZT4y1M75y

**Comment by giter** [comment]
I'm sure that the connection is successful when using opc-quick-client or konimarti/opc

**Comment by huskar-t** [comment]
@giter Describe your environment and setup

**Comment by huskar-t** [comment]
@giter What opc da server are you using?

**Comment by huskar-t** [comment]
@giter Try compiling the program into 32-bit.
`GOARCH=386`

**Comment by giter** [comment]
build **example/browse** with **GOARCH=386**  and run it remote ,  result:

![snapte](https://github.com/user-attachments/assets/e4d3e987-8c22-438d-bbb3-97ba604532f3)

**Comment by huskar-t** [comment]
@giter Can you try running in administrator account?
